### PR TITLE
File browser now allows typing a path

### PR DESCRIPTION
**Pull request by VinceVal**
Also added showBrowseButton option to $.fn.fileBrowser().

### CODE REVIEW

File: gui/slick/js/browser.js [hunk 1]
```diff
@@ -34,7 +35,20 @@
             data = $.grep(data, function (value) {
                 return i++ !== 0;
             });
-            $('<h2>').text(first_val.current_path).appendTo(fileBrowserDialog);
+
+            $('<input type="text" class="form-control input-sm">')
```
Comment (OmgImAlexis) [inline gui/slick/js/browser.js:39]: This should be something like this.

```
$('input[type="text"].form-control.input-sm">')
```

File: gui/slick/js/browser.js [hunk 2]
```diff
@@ -166,13 +180,19 @@
...
-            var initialDir = options.field.val() || (options.key && path) || '';
-            var optionsWithInitialDir = $.extend({}, options, {initialDir: initialDir});
-            $(this).nFileBrowser(callback, optionsWithInitialDir);
-            return false;
-        }));
+        options.field.addClass('fileBrowserField');
+        if (options.showBrowseButton) {
+            // append the browse button and give it a click behaviour
+            options.field.after(
+                $('<input type="button" value="Browse&hellip;" class="btn btn-inline fileBrowser">').click(function () {
```
Comment (OmgImAlexis) [inline gui/slick/js/browser.js]: Same for here.

This should also be using `on('click', function() {` instead of `click(function () {`.

**Comment by fernandog** [comment]
What if path does not exist?

**Comment by VinceVal** [comment]
@fernandog If the path doesn't exist the AJAX callback will climb the tree until it finds one that does.

**Pull request merged**